[feature] support allow-pypi-requests in environment yaml (#836)
conda/conda-lock: README.md

The location of this file can be determined with `python -c 'from conda_lock._vendor.poetry.locations import CONFIG_DIR; print(CONFIG_DIR)'`

> Private repositories will be used in addition to `pypi.org`. It is possible to [disable `pypi.org` entirely](#disabling-pypiorg) in the `environment.yml` by setting `allow-pypi-requests: false`.
> 

Reviewer: Let's avoid specifying the `environment.yml` config twice.

```suggestion
By default, private repositories will be used in addition to `pypi.org`, but it is also possible to [disable `pypi.org` entirely](#disabling-pypiorg).
```

Private repositories will be used in addition to `pypi.org`. For projects using `pyproject.toml`, it is possible to [disable `pypi.org` entirely](#disabling-pypiorg).

> #### Disabling pypi.org

Author: I've pushed a change to the README. Does this look a little better?
